src/msw/spinctrl.cpp

```diff
@@ -366,28 +334,36 @@ bool wxSpinCtrl::Create(wxWindow *parent,
     m_wndProcBuddy = (WXFARPROC)wxSetWindowProc(GetBuddyHwnd(),
                                                 wxBuddyTextWndProc);
 
+    // associate the text window with the spin button
+    (void)::SendMessage(GetHwnd(), UDM_SETBUDDY, (WPARAM)m_hwndBuddy, 0);
+
     // set up fonts and colours  (This is nomally done in MSWCreateControl)
     InheritAttributes();
     if (!m_hasFont)
         SetFont(GetDefaultAttributes().font);
 
-    // set the size of the text window - can do it only now, because we
-    // couldn't call DoGetBestSize() before as font wasn't set
-    if ( sizeText.y <= 0 )
+    // the given width is the total width for both controls
+    // adjust invalid width for text control
+    wxSize sizeCtrl(size);
+    if ( sizeCtrl.x <= 0 )
+    {
+        // DEFAULT_ITEM_WIDTH is the default width for the text control
+        sizeCtrl.x = FromDIP(DEFAULT_ITEM_WIDTH);
```

Comment:
I've changed this to preserve (almost) the same default size as before the changes, i.e. add (roughly) the button width (it was `width + 1` before, it is `width - overlap` now).